core/state/dump.go

```diff
@@ -49,6 +49,13 @@ type Dump struct {
 // iterativeDump is a 'collector'-implementation which dump output line-by-line iteratively
 type iterativeDump json.Encoder
 
+// IteratorDump is a
+type IteratorDump struct {
+	Root     string                         `json:"root"`
```

Comment:
```suggestion
	Root     common.Hash                         `json:"root"`
```

Reply:
The other Dump method all returns as `string`, IMO, consistency is more important.

Reply:
You are correct. This is how the current dump looks: 
```
{"balance":"1000000000000000000000","nonce":0,"root":"56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","codeHash":"c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470","address":"0xc518799a5925576213e21896e0539abb85b05ae3"}
```
No `0x`-prefixed `codeHash`, nor `code`, nor `root`.